src/Liquifier.sol

```diff
@@ -180,9 +181,11 @@ contract Liquifier is Initializable, UUPSUpgradeable, OwnableUpgradeable, Pausab
     function swapEEthForStEth(uint256 _amount) external whenNotPaused nonReentrant {
         if (_amount > lido.balanceOf(address(this))) revert NotEnoughBalance();
         if (_amount > liquidityPool.eETH().balanceOf(msg.sender)) revert NotEnoughBalance();
+        if (!roleRegistry.hasRole(EETH_STETH_SWAPPER, msg.sender)) revert IncorrectRole();
 
         IERC20(address(liquidityPool.eETH())).safeTransferFrom(msg.sender, address(this), _amount);
         IERC20(address(lido)).safeTransfer(msg.sender, _amount);
+        liquidityPool.withdraw(address(liquidityPool), _amount-1);
```

Comment:
what would be the clean way to handle `-1`? 
We can do diff of stETH balance before/after its transfer and transfer that amount